### PR TITLE
spec: add s3 configuration into s3 source

### PR DESCRIPTION
**Pull request by xiang90**
/cc @hongchaodeng @hasbro17 

[skip ci]

### CODE REVIEW

File: pkg/spec/backup.go [hunk 1]
```diff
@@ -73,7 +73,24 @@ type PVSource struct {
...
+	// The name of the AWS S3 bucket to store backups in.
+	//
+	// S3Bucket overwrites the default etcd operator wide bucket.
+	S3Bucket string `json:"s3Bucket,omitempty"`
+
+	// The name of the secret object that stores the AWS credential file.
+	// The file name must be 'credentials'.
+	//
+	// AWSCredentialSecret overwrites the default etcd operator wide AWS credential.
+	AWSCredentialSecret string `json:"awsCredentialSecret,omitempty"`
```
Comment (hongchaodeng) [inline pkg/spec/backup.go]: Based on past experience, it would be better to combine both credential and config into one secret object.

Comment (xiang90) [inline pkg/spec/backup.go]: sounds good to me. I also feel so. 

are we going to deprecate the old behavior in etcd operator?

Comment (hongchaodeng) [inline pkg/spec/backup.go]: Let's put the spec in first. Then we will think about upgrade (maybe not in next release 0.2.7). Then we think about deprecation.

**Comment by xiang90** [comment]
@hongchaodeng updated.

**Comment by hasbro17** [comment]
LGTM

**Comment by hongchaodeng** [comment]
lgtm

**Pull request merged**